frappe/public/js/frappe/router.js

```diff
@@ -334,7 +334,7 @@ frappe.router = {
 				return null;
 			} else {
 				a = String(a);
-				if (a && a.match(/[%'"]/)) {
+				if (a && a.match(/[%'"\s\t]/)) {
 					// if special chars, then encode
 					a = encodeURIComponent(a);
 				}
```

Comment:
I'm not sure about the implications, but would there be any downsides in just URL-encoding all strings for routing anyway?

```suggestion
				a = encodeURIComponent(a);
```

Reply:
This was done to keep URLs pretty when there was a space in the route.

For e.g., `item/Tennis Ball` is a valid route and works in the browser. If we encode it, it becomes `item/Tennis%20Ball` which is ugly.

However, when the route was like `item/Tennis Ball 5%', it had to be encoded.

The change in the PR defeats the original purpose. However, we can try and replace spaces with `+` to keep them pretty.

Reply:
Ah, gotcha. Seems like there's some ideas on how [spaces in a URL should be handled](https://stackoverflow.com/a/29948396)?

Reply:
@Alchez Wow, learned something new.

@prssanna Maybe we should follow the convention then?